File: app/src/main/java/com/softaai/designpatterns/java/creational/prototype/PrototypePatternDemo.java
package com.softaai.designpatterns.java.creational.prototype;

/**
 * Created by Amol Pawar on 02-04-2019.
 * softAai Apps
 */
public class PrototypePatternDemo {

    public static void main(String[] args){
        ShapeCache.loadCache();

        Shape clonedShape = ShapeCache.getShape("1");
        System.out.println("Shape : " + clonedShape.getType());

        Shape clonedShape2 = ShapeCache.getShape("2");
        System.out.println("Shape : " + clonedShape2.getType());

        Shape clonedShape3 = ShapeCache.getShape("3");
        System.out.println("Shape : " + clonedShape3.getType());
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/creational/prototype/PrototypePatternDemo.kt
package com.softaai.designpatterns.kotlin.creational.prototype


/**
 * Created by Amol Pawar on 02-04-2019.
 * softAai Apps
 */
object PrototypePatternDemo{
    @JvmStatic
    fun main(args: Array<String>){
        ShapeCache.loadCache()

        val clonedShape : Shape = ShapeCache.getShape("1")
        println("Shape : " + clonedShape.getType())

        val clonedShape2 : Shape = ShapeCache.getShape("2")
        println("Shape : " + clonedShape2.getType())

        val clonedShape3 : Shape = ShapeCache.getShape("3")
        println("Shape : " + clonedShape3.getType())
    }
}